### DISCUSSION TITLE
New update ?

### DISCUSSION THREAD

**Discussion by JohnBoy200** [valinet/ExplorerPatcher#442] [Q&A] [answered]
Once again thanks for this program much appreciated.

It prompted me there was a new update 318.38.6 so I let it auto install but it installs 318.38.2 again

I've downloaded and maually installed 318.38.6 which works fine but it still says there is a new update of 318.38.6 if I let it auto install it installs 318.32 again, so I've downloaded 318.38.6 again and installed it manually.

Not sure whats going on.

**Comment by BraINstinct0** [comment, marked as answer]
Currently update check only checks for releases not prereleases. Since 318.38.2 is the latest stable release, that'll be installed. Summary: intended behavior, though it may be addressed in the future.

**Reply by JohnBoy200** [reply]
Thanks for the reply. I should've realised that lol
Keep up the good work.

**Reply by BraINstinct0** [reply]
@valinet is the sole developer - all credits to the dev ;)

**Reply by valinet** [reply]
38.9 adds option to include pre-releases when checking for updates.

**Reply by JohnBoy200** [reply]
When I use the built in check for update is it possible to show what the update version is that is going to be download rather then just say there is a new update.
I'm using a pre-release 38.8 and now 38.9, it still says there is a new update but as pointed out to me I now know it looks at the latest version to download which is 38.2
38.2 for me isnt  working on W11 22000.348 so if I agree to update I only see the version that it's going to update to as it it updates.
Not a major problem as I keep the versions or redownload the pre-release that works.

**Reply by valinet** [reply]
I don't remember if 38.8 included this, but 38.9 or 39.0 or whatever is the latest one should definitely include an option where you can have it check among the pre-releases as well. This way, the confusion should pretty much be gone, so a new version actually means a newer version. Clicking the notification also takes you to the release notes. Due to the lightness of the technical implementation, telling the version in the notification would take much more resources, so to say, but the above alleviates it to the point where I hardly believe is need. Like, the actual version number does not matter that much; what it really mattered to you, as you said, is knowing whether that really is a new version, and that is addressed by the what I described above. Thanks.

**Reply by valinet** [reply]
So yeah, now I see I already said this above, so enable: 

![image](https://user-images.githubusercontent.com/6503598/143233263-3912d662-f5ac-49eb-8250-8e99a150a9d8.png)

And it should work for your use case, it will also check among pre-releases.

**Reply by JohnBoy200** [reply]
As I said not a major problem and if it would be resource hungry best not to bother. Yes indeed the pre-release option is there but not recommended so I think I'll continue to download manually after I've looked here at least until I see a latest release that works on 348.

Thanks again.

**Reply by valinet** [reply]
It will come in 5 minutes. It's not recommended generally, it's recommended when it makes sense. Anyway, I will push a release in 5 minutes and all should be good.

**Reply by JohnBoy200** [reply]
Thank you, now it won't keep nagging me to update when I startup.

You've really done a good job, it's made W11 so much better.

**Comment by Anixx** [comment]
Well, actually this policy makes the updates feature useless, at least for me. Especially, given it can downgrade the current version. I still have to to to GitHub to see if there is a newer version and manually download the updater.

**Reply by valinet** [reply]
38.9 adds option to include pre-releases when checking for updates.

**Comment by Anixx** [comment]
Maybe @valinet can consider adding another button for pre-release update? Or simply reconsider the policy.

**Reply by valinet** [reply]
What's to reconsider?
* If I push all updates to all users, not only I cannot make commits because some get triggered and annoyed by them being too frequent, there is also the risk of pushing an update that makes a regression and then the storm begins.. tickets, issues, reports, rushing to fix etc.
* If I make tiered releases, where most stuff gets built and published as a pre-release, as you previously asked, if you are so kind to remember, so that you do not have to go though the hassle of building from source yourself or even downloading the artifacts, and when builds are sufficiently tested they are pushed to all users via releases, it's still not good apparently because it quote "makes the updates feature useless, at least for me".

**Reply by JohnBoy200** [reply]
I think its best as it is. Good idea not to release it until you're good and ready.

People can always download it manually as I mistakenly did.

**Reply by valinet** [reply]
Yeah, if GitHub offered an endpoint like `https://github.com/valinet/ExplorerPatcher/releases/latest` but for pre-releases, one could set `UpdateURL` in `HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher` to that and then it would use that as an update server and get pre-releases. Idk, maybe some solution could be found regarding that as well, much more useful if people interested would look into that than the easy route of "this policy makes the updates feature useless".

**Reply by BraINstinct0** [reply]
~~First draft(more of just me rambling thoughts until I can come up with a solution):~~
Good Evil Google points me at this:  https://stackoverflow.com/a/67689318
Quick, dirty JSON for your repo's releases is [here](https://api.github.com/repos/valinet/ExplorerPatcher/releases). Though, it seems to use github's API which has rate limits. I don't think that'll be an issue here.. If this query is done from the user's side(unauthenticated) [60 times per hour per IP seems to be the limit](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting)
Parsing should be done but at least this can be a point to start.

Parsing that JSON.. seems... doable? But I'm mostly a Python person, so I really don't know how it will scale to C/C++.

**Reply by valinet** [reply]
@BraINstinct0 Thanks. It's doable, there are APIs in the Windows Runtime for example, there are previous working examples of mine as well, like https://github.com/valinet/ThunderbirdToasts/blob/master/src/native-app/thunderbird_toasts/thunderbird_toasts.c#L551, someone just needs to write that. I have other stuff to work on at the moment. It's just another complication to add to this... oh, well...

Or, some dirty quick solution, have an option that enables this, where like, I download the first 500 bytes or whatever, looking for the first `"browser_download_url"` in there, as that seems to be the URL of the most recent file, either pre-release, either release, it's much easier to implement. The order of the returned data seems to be most recent to oldest. This, actually more likely to come sooner rather than later.

Anyway, I will look into it. Thanks for the pointers, it helps with the thought process a fair share when you guide someone a bit rather than throw yet something else on the table.

**Reply by valinet** [reply]
38.9 adds option to include pre-releases when checking for updates.

**Reply by Anixx** [reply]
Super!
